server/repository/src/db_diesel/stock_line.rs

```diff
@@ -160,15 +161,27 @@ impl<'a> StockLineRepository<'a> {
 type BoxedStockLineQuery = IntoBoxed<
     'static,
     LeftJoin<
-        LeftJoin<LeftJoin<InnerJoin<stock_line::table, item::table>, location::table>, name::table>,
+        LeftJoin<
+            LeftJoin<
+                InnerJoin<stock_line::table, InnerJoin<item_link::table, item::table>>,
+                location::table,
+            >,
+            name::table,
+        >,
         barcode::table,
     >,
     DBType,
 >;
 
+allow_tables_to_appear_in_same_query!(stock_line, item_link);
+allow_tables_to_appear_in_same_query!(item, item_link);
+allow_tables_to_appear_in_same_query!(location, item_link);
+allow_tables_to_appear_in_same_query!(name, item_link);
+allow_tables_to_appear_in_same_query!(barcode, item_link);
+
 fn create_filtered_query(filter: Option<StockLineFilter>) -> BoxedStockLineQuery {
     let mut query = stock_line_dsl::stock_line
-        .inner_join(item_dsl::item)
+        .inner_join(item_link_dsl::item_link.inner_join(item_dsl::item))
```

Comment:
Instead of returning item as before, we join on item_link and get the item related to the item_link.item_id, returned as a tuple. Then we get a tuple of `(ItemLinkRow, ItemRow)` out of the query. In `to_domain()` we just ditch the link and return the item as if nothing weird happened.